app/controllers/marine_controller.rb

```diff
@@ -1,6 +1,17 @@
 class MarineController < ApplicationController
 
+  #Calculated stats
+  before_action :coverage
```

Comment:
It will make the code more structured I believe.
The other way of doing it is just calling the methods inside the action, but supposing we are going to add something in the index action or even create other actions, this will keep things cleaner